backend/assignments/2/src/main/java/com/caching/assignment2/controller/GeoController.java

```diff
@@ -0,0 +1,80 @@
+package com.caching.assignment2.controller;
+
+import com.caching.assignment2.exception.GeoServiceException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import com.caching.assignment2.dto.ForwardGeocodingRequestDTO;
+import com.caching.assignment2.dto.ReverseGeocodingRequestDTO;
+import com.caching.assignment2.dto.ForwardGeocodingResponseDTO;
+import com.caching.assignment2.dto.ReverseGeocodingResponseDTO;
+import com.caching.assignment2.service.GeoService;
+
+/**
+ * RESTful controller class for handling geocoding-related endpoints.
+ */
+@RestController
+public class GeoController {
+
+    private final GeoService geoService;
+
+    /**
+     * Constructor for GeoController.
+     *
+     * @param geoService An instance of GeoService for handling geocoding requests.
+     */
+    @Autowired
+    public GeoController(GeoService geoService) {
+        this.geoService = geoService;
+    }
+
+    /**
+     * Endpoint for forward geocoding.
+     *
+     * @param address The address to be geocoded.
+     * @return ResponseEntity containing the ForwardGeocodingResponseDTO or an error response.
+     * @throws GeoServiceException if an error occurs during geocoding.
+     */
+    @GetMapping("/geocoding")
+    public ResponseEntity<ForwardGeocodingResponseDTO> forwardGeocoding(@RequestParam String address) {
+        try {
```

Comment:
1. All the validations can be abstracted out in methods. They can also be handled in service.
2. The exception throw/catch can also be done in the service.